lib/vsc/install/shared_setup.py

```diff
@@ -174,6 +175,161 @@ def run(self):
+        testsuites = ScanningLoader.loadTestsFromModule(self, module)
+
+        test_filter = getattr(__builtin__,'__test_filter')
+
+        res = testsuites
+
+        if test_filter['module'] is not None:
+            name = module.__name__
+            if name in test_filter['allowmods']:
+                # a parent name space
+                pass
+            elif re.search(test_filter['module'], name):
+                if test_filter['function'] is not None:
+                    res = filter_testsuites(testsuites)
+                # add parents (and module itself)
+                pms = name.split('.')
+                for pm_idx in range(len(pms)):
+                    pm = '.'.join(pms[:pm_idx])
+                    if not pm in test_filter['allowmods']:
+                        test_filter['allowmods'].append(pm)
+            else:
+                res = type(testsuites)()
+
+        return res
+
+
+class VscTestCommand(TestCommand):
+    """
+    The cmdclass for testing
+    """
+
+    # make 2 new 'python setup.py test' options available:
+    #    --test-filterf and --test-filertm
+    user_options = TestCommand.user_options + [
+        ('test-filterf=', 'f', "Regex filter on test function names"),
+        ('test-filterm=', 'F', "Regex filter on test (sub)modules"),
+    ]
+
+    TEST_LOADER_CLASS = VscScanningLoader
+
+    def initialize_options(self):
+        """
+        Add attributes for new commandline options and set test_loader
+        """
+        TestCommand.initialize_options(self)
+        self.test_filterm = None
+        self.test_filterf = None
+        self.test_loader = '%s:%s' % (self.TEST_LOADER_CLASS.TEST_LOADER_MODULE, self.TEST_LOADER_CLASS.__name__)
+        log.info("test_loader set to %s" % self.test_loader)
+
+    def setup_sys_path(self):
+        """
+        Prepare sys.path to be able to
+            use the modules provided by this package (assumeing they are in 'lib')
+            use any scripts as modules (for unittesting)
+            use the test modules as modules (for unittesting)
+        Returns a list of directories to cleanup
+        """
+        cleanup = []
+
+        # determine the base directory of the repository
+        # we will assume that the tests are called from 
+        # a 'setup.py' like file in the basedirectory
+        # (but could be called anything, as long as it is in the basedir) 
+        setup_py = os.path.abspath(sys.argv[0])
+        log.info('run_tests from %s' % setup_py)
+        base_dir = os.path.dirname(setup_py)
+
+        # make a lib dir to trick setup.py to package this properly
+        # and git ignore empty dirs, so recreate it if necessary
+        lib_dir = os.path.join(base_dir, 'lib')
+        if not os.path.exists(lib_dir):
+            os.mkdir(lib_dir)
+            cleanup.append(lib_dir)
+
+        test_dir = os.path.join(base_dir, DEFAULT_TEST_SUITE)
+        if os.path.isdir(test_dir):
+            sys.path.insert(0, test_dir)
```

Comment:
no, should add testdir for modules in test dir that are no unittests but are used in unittests